noir-projects/noir-protocol-circuits/crates/blob/src/blob_batching_public_inputs.nr

```diff
@@ -0,0 +1,280 @@
+};
+
+pub struct BatchingBlobCommitment {
+    pub point: BLSPoint,
+    pub compressed: [u8; 48], // TODO(MW): get Q::num_bytes from somewhere, rather than hardcoded 48
+}
+
+impl BatchingBlobCommitment {
+    // The compressed form is a BLS12Fq field encoded as 2 BN fields.
+    // The first is the first 31 bytes, and the second is the next 17 bytes:
+    pub fn to_compressed_fields(self) -> [Field; 2] {
+        [
+            // field 0 is bytes 0..31
+            field_from_bytes::<31>(array_splice(self.compressed, 0), true),
+            // field 1 is bytes 31..48
+            field_from_bytes::<17>(array_splice(self.compressed, 31), true),
+        ]
+    }
+}
+
+impl Eq for BatchingBlobCommitment {
+    fn eq(self, other: Self) -> bool {
+        (self.point.eq(other.point)) & (self.compressed.eq(other.compressed))
+    }
+}
+
+impl Empty for BatchingBlobCommitment {
+    fn empty() -> Self {
+        Self { point: BLSPoint::point_at_infinity(), compressed: [0; 48] }
+    }
+}
+
+// TODO(MW): get_flags() and compress_to_bytes() will eventually be part of BigCurve.
+
+// Ideally we would impl like:
+// impl From<BLSPoint> for BlobCommitment {
+//     }
+// }
+// impl BLSPoint {
+//     fn get_flags(point: BLSPoint) -> ([bool; 3], Q) {
+//     }
+//     fn compress_to_bytes(self) -> [u8; 48] {
+//     }
+// }
+
+// So, e.g., can just call point.compress_to_bytes() in future. For now:
+pub fn compress_to_blob_commitment(point: BLSPoint) -> (BatchingBlobCommitment) {
+    let (flags, x) = get_flags(point);
+    // TODO(MW): use built in .to_be_bytes() in bignum 0.7.0
+    let mut compressed = bignum_limbs_to_be_bytes::<_, 381>(x.limbs);
+    let most_sig_bits = byte_to_bits_be(compressed[0]);
+    // TODO(MW): May not need to assert this since Bignum may check it's 381 bits?
+    for i in 0..3 {
+        assert_eq(most_sig_bits[i], 0, "Invalid BLS12-381 x coordinate given to compress().");
+    }
+    // Flip some bits (NB: this is probably not efficient but just easier to log/visualise for now)
+    let mut flip = 0;
+    // flags = [is_compressed, is_infinity, is_greater]
+    // if (is_compressed) { flip most sig bit in u8 => byte |= 2**7 => byte |= 2 << 6 }
+    // if (is_infinity) { flip next most sig bit in u8 => byte |= 2**6 => byte |= 2 << 5 }
+    // if (is_compressed) { flip next most sig bit in u8 => byte |= 2**5 => byte |= 2 << 4 }
+    for i in 0..3 {
+        if flags[i] {
+            flip += 2 << (6 - i);
+        }
+    }
+    compressed[0] |= flip;
+    BatchingBlobCommitment { point, compressed }
+}
+
+fn get_flags(point: BLSPoint) -> ([bool; 3], Q) {
+    let x = point.x;
+    let y = point.y;
+    let is_greater = y > -y;
+    let is_infinity = point.is_infinity;
+    let is_compressed = true;
+    let flags = [is_compressed, is_infinity, is_greater];
+    (flags, x)
+}
+
+// Will be included as .to_be_bytes() in bignum 0.7.0 TODO(MW): remove when bumped
+fn bignum_limbs_to_be_bytes<let N: u32, let MOD_BITS: u32>(
```

Comment:
Copied from https://github.com/noir-lang/noir-bignum/pull/157/ and is included in `bignum` 0.7.0, which is breaking for `bigcurve` and protocol circuits. See #13609